src/main/java/use_cases/turn_end_use_case/TurnEndInteractor.java

```diff
@@ -0,0 +1,111 @@
+package use_cases.turn_end_use_case;
+
+import java.util.List;
+import java.util.ArrayList;
+
+import entities.GameState;
+import entities.Player;
+import entities.cardEffects.CardEffect;
+import entities.cardEffects.CreatureStatsEffect;
+import entities.cardEffects.PlayerStatsEffect;
+import entities.cards.Card;
+import entities.cards.CreatureCard;
+import entities.cards.StructureCard;
+
+public class TurnEndInteractor implements TurnEndInputBoundary {
+
+    private final GameState gameState;
+    private final TurnEndOutputBoundary turnEndPresenter;
+
+    public TurnEndInteractor(GameState gameState, TurnEndOutputBoundary turnEndPresenter) {
+        this.gameState = gameState;
+        this.turnEndPresenter = turnEndPresenter;
+    }
+    
+    @Override
+    public void passTurn() {
+        gameState.switchPlayer();
+        // 1. pass player's name to presenter
+        // 2. whatever other effects that are needed
+    }
+
+    @Override
+    public TurnEndResponseModel triggerEndTurnEffects() {
+        Player currentPlayer = gameState.getCurrentPlayer();
+        Player nextPlayer = gameState.getOpposingPlayer();
+
+
+        // // 1. Check if the current player has a structure in play
+        // // 1a. Yes, lets trigger the event
+        if (currentPlayer.getStructure() != null) {
+            // Trigger the effects needed
+            StructureCard currentStructure = currentPlayer.getStructure();
+            List<CreatureCard> allTargets = new ArrayList<>();
+
```

Comment:
I believe that you need to check to make sure the structureCard trigger event is the same as the Game_Event.TURN_END.